app/controllers/filters_controller.rb

```diff
@@ -69,7 +69,7 @@ def find_role
 
   def resource_base
     @resource_base ||= if @role.present?
-                         @role.filters.authorized(current_permission)
+                         Filter.authorized_as(User.current, current_permission)
```

Comment:
as suggested on IRC, `Filter.authorized(current_permission)` would be nicer if it works